Feature/availability label

Adamik10 opened the pull request:
#### What does this PR do?
- Adds Availability Label component 
- Introduces dpl-design-system as an npm package

#### Where should the reviewer start?
- File by file

#### How can the reviewer see this in action?
- By following our README.md file in order to run the project in the development environment

#### Any background context you want to provide?
The availability label design was created in the dpl-design-system repository, and now we had to create the corresponding react component.

#### Is this related to processing of personal data?
No.

#### What are the relevant tickets?
N/A

#### Screenshots (if appropriate)
-

#### Questions for the reviewer:
- Why have all the checks on PR failed?
- What do you think of the way I solved this task? 
- Would you have done something differently? 
- If applicable, why and how would you solve it?

Adamik10 committed:
Added an overarching css file
- this file is copied from dpl-design-system repository
- the solution is temporary, as we needed a way to utilize the styling from dpl-design-system
- in the future, we would like to be able to install dpl-design-system as a dev dependancy for this project

Adamik10 committed:
Adjusted the README.md file
- there was an inprecision in the readme that posed a problem for setting up the project
- added an alternative way to setup the development environment

Adamik10 committed:
Created Availability Label Component
- created the react component + story dev files + scss
- the scss file is currently used to style the check icon file, which is a temporary solution for now
- in the future, this check is going to be imported from the dpl-design system
- also added color variables into the components.scss

Adamik10 committed:
Added Utils folder
- this folder is used for utility components that do not have a story of their own, such as links

Adamik10 committed:
Link functionality added to Availability Label
- if a link prop is passed to it, the component will get wrapped in a link

Adamik10 committed:
Added icons from dpl-design-system
- Availability label now shows the check icon if it is in the "selected" state

Adamik10 committed:
Updated README.md + added .yarnrc
- the README.md was incomplete in regards to no docker dev setup
- added a step that allows an error-free login through adgangsplatformen
- .yarnrc file is used in order to be able to use npm package from dpl-design-system

spaceo commented on src/components/availability-label/check.jsx:
```diff
@@ -0,0 +1,36 @@
+ */
+function Check({ classes }) {
+  return (
+    <svg
```
I think we should move svg's to separate files to reduce code "clutter"

spaceo commented on src/components/utils/link.jsx:
```diff
@@ -0,0 +1,32 @@
+ * @param {boolean} props.newTab
+ * @returns {ReactNode}
+ */
+const Link = ({ href, children, newTab }) => {
```
You could consider following the convention we talked about earlier about clear usage of booleans and call the prop: "shouldOpenNewTab"

Adamik10 replied:
nice spotted! I named it isNewTab, for ease of use :)

spaceo commented on package.json:
```diff
@@ -99,6 +100,7 @@
     "@reach/auto-id": "^0.6.1",
     "@reach/dialog": "^0.5.4",
     "@reduxjs/toolkit": "^1.4.0",
+    "@reload/dpl-design-system": "0.0.0-0ede2ac4e55c0a4ce4c960210dcf0a0ddb4fc7af",
```
Do we have a clear strategy of which packages to use here?
Branch names is more human understandable, I guess.

Adamik10 replied:
yes, that is true, but we need to be able to distinguish between packages of the same branch on each push. 
That's why we take the commit hash (i believe)

Adamik10 replied:
mmm yeah. It is currently not very readable for a human eye. We could maybe add the branch name before the commit id

spaceo commented on src/components/availability-label/availability-label.scss:
```diff
@@ -0,0 +1,15 @@
+/* stylelint-disable */
+.check-icon {
```
This will be a reoccuring question :)
Is this styling necessary? It is not defined i dpl-design-system?

Adamik10 replied:
good point! I'll implement it in the design system instead :)

spaceo commented on .nvmrc:
```diff
@@ -0,0 +1 @@
+v12.13.0
```
Need new line here :)

Adamik10 replied:
thx :)

spaceo reviewed (changes requested):
Looks good! Have a few remarks.

Adamik10 commented on src/components/availability-label/availability-label.jsx:
```diff
@@ -0,0 +1,78 @@
+ * @returns {ReactNode}
+ */
+const AvailabilityLabel = ({ manifestText, availabilityText, state, link }) => {
+  function availableClass(isAvailable) {
+    return isAvailable !== "unavailable" ? "success" : "alert";
+  }
+
+  function selectedClass(currentState) {
+    return currentState === "selected" ? "selected" : "unselected";
+  }
+
+  function selectedClassTriangle(currentState) {
+    return currentState === "selected" ? "none" : "xsmall";
+  }
+
+  function checkIconClasses(currentState) {
+    return currentState === "selected" ? "selected" : "";
+  }
```
yes! done that :)

Adamik10 committed:
Update prop naming for Booleans in Availability Label component & link

Props and variables that hold Boolean values should be named according
to the common standard, which was not the case for Availability Label
component and link utility.

Availability Label story was set up with a flaw, where the link prop
was defined on the label in "selected" state. This now reflects reality,
and the link prop is undefined for this state.

Adamik10 committed:
Use clsx library for dynamic classes in Availability Label component

Clsx makes it easier to conditionally apply classes, and increases
readability of the code.

Adamik10 committed:
Update DPL design system library and its usage in the repository

The package was outdated. the dpl-design-system repository has released
a newer version of this package.

Use the dynamic import of the library instead of copying the css build
file into the project. This avoids extra work adding newer versions.

Adamik10 committed:
Remove Availability Label style, and copied asset files

We should ideally not introduce any styling to the components in this
project. All the styles and assets are retrieved from dpl-design-system
library.
